src/index/am_scan.rs

```diff
@@ -42,18 +72,35 @@ pub fn scan_next(scanner: &mut Scanner, handle: Handle) -> Option<Pointer> {
                         Err((_, VbaseError::InvalidVector)) => bad_service_invalid_vector(),
                         Err((_, VbaseError::InvalidSearchOptions { reason: _ })) => unreachable!(),
                     };
-                    *scanner = Scanner::Vbase { vbase };
+                    *scanner = Scanner::Vbase {
+                        vbase,
+                        threshold: *threshold,
+                        recheck: *recheck,
+                    };
                 }
             }
         } else {
             *scanner = Scanner::Empty {};
         }
     }
-    match scanner {
+    let (result, threshold, recheck) = match scanner {
         Scanner::Initial { .. } => unreachable!(),
-        Scanner::Basic { basic, .. } => basic.next(),
-        Scanner::Vbase { vbase, .. } => vbase.next(),
-        Scanner::Empty {} => None,
+        Scanner::Basic {
+            basic,
+            threshold,
+            recheck,
+        } => (basic.next(), *threshold, *recheck),
+        Scanner::Vbase {
+            vbase,
+            threshold,
+            recheck,
+        } => (vbase.next(), *threshold, *recheck),
+        Scanner::Empty {} => (None, None, false),
+    };
+    match (result, threshold) {
+        (Some((_, ptr)), None) => Some((true, recheck, ptr)),
+        (Some((distance, ptr)), Some(t)) => Some((distance.to_f32() < t, recheck, ptr)),
+        _ => None,
```

Comment:
It should always continue to yield `None` when exhausted, in order to avoid this situation:
```
IndexNext(node) -> Some((pointer, recheck))
IndexNext(node) -> None
IndexNext(node) -> Some((pointer, recheck))
IndexNext(node) -> None
// ...
```
You can replace `scanner` in this way:
```rust
let scanner = std::mem::replace(scanner, Scanner::Empty {});
am_scan::scan_release(scanner);
```